app/src/main/java/de/grobox/transportr/trips/search/TripViewHolder.kt

```diff
@@ -36,39 +36,29 @@ import de.schildbach.pte.dto.Trip
 import de.schildbach.pte.dto.Trip.Individual
 import de.schildbach.pte.dto.Trip.Public
 
-internal class TripViewHolder(private val root: View) : RecyclerView.ViewHolder(root) {
+internal class TripViewHolder(private val v: View) : BaseViewHolder(v) {
 
-    private val context = root.context
-    private val fromTimeRel: TextView = root.findViewById(R.id.fromTimeRel)
-    private val fromTime: TextView = root.findViewById(R.id.fromTime)
-    private val fromLocation: TextView = root.findViewById(R.id.fromLocation)
-    private val fromDelay: TextView = root.findViewById(R.id.fromDelay)
-    private val toTime: TextView = root.findViewById(R.id.toTime)
-    private val toLocation: TextView = root.findViewById(R.id.toLocation)
-    private val toDelay: TextView = root.findViewById(R.id.toDelay)
-    private val warning: View = root.findViewById(R.id.warning)
-    private val lines: ViewGroup = root.findViewById(R.id.lines)
-    private val duration: TextView = root.findViewById(R.id.duration)
+    private val fromTimeRel: TextView = v.findViewById(R.id.fromTimeRel)
+    private val fromLocation: TextView = v.findViewById(R.id.fromLocation)
+    private val toLocation: TextView = v.findViewById(R.id.toLocation)
+    private val warning: View = v.findViewById(R.id.warning)
+    private val lines: ViewGroup = v.findViewById(R.id.lines)
+    private val duration: TextView = v.findViewById(R.id.duration)
 
     fun bind(trip: Trip, listener: OnTripClickListener) {
         // Relative Departure Time
         setRelativeDepartureTime(fromTimeRel, trip.firstDepartureTime)
 
-        // Departure Time
+        // Departure Time and Delay
         val firstLeg = trip.legs[0]
         if (firstLeg is Public) {
-            fromTime.text = getTime(context, firstLeg.getDepartureTime(true))
```

Comment:
Did you check that what we do in `setDepartureTimes()` is equivalent to what happens in `firstLeg.getDepartureTime(true)`?

Reply:
It is equivalent to what happened before in this line and in lines 66-71.